packages/storage-web-api/src/controllers/post-website-controller.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as ApiContracts from 'api-contracts';
+import { GuidGenerator, ServiceConfiguration } from 'common';
+import { inject, injectable } from 'inversify';
+import _ from 'lodash';
+import { ContextAwareLogger } from 'logger';
+import { HttpResponse, WebApiErrorCodes, ApiController, WebsiteProvider, PageProvider } from 'service-library';
+import * as StorageDocuments from 'storage-documents';
+import { createPageApiResponse, PageDocumentResponseConverter } from '../converters/page-document-response-converter';
+import { createWebsiteApiResponse, WebsiteDocumentResponseConverter } from '../converters/website-document-response-converter';
+
+@injectable()
+export class PostWebsiteController extends ApiController {
+    public readonly apiVersion = '1.0';
+
+    public readonly apiName = 'storage-web-api-get-website';
+
+    public constructor(
+        @inject(ServiceConfiguration) protected readonly serviceConfig: ServiceConfiguration,
+        @inject(ContextAwareLogger) logger: ContextAwareLogger,
+        @inject(GuidGenerator) private readonly guidGenerator: GuidGenerator,
+        @inject(WebsiteProvider) private readonly websiteProvider: WebsiteProvider,
+        @inject(PageProvider) private readonly pageProvider: PageProvider,
+        private readonly isValidWebsite: typeof ApiContracts.isValidWebsite = ApiContracts.isValidWebsite,
+        private readonly convertWebsiteDocumentToResponse: WebsiteDocumentResponseConverter = createWebsiteApiResponse,
+        private readonly convertPageDocumentToResponse: PageDocumentResponseConverter = createPageApiResponse,
+    ) {
+        super(logger);
+    }
+
+    public async handleRequest(): Promise<void> {
+        this.logger.setCommonProperties({ source: 'postWebsiteRESTApi' });
+
+        const websiteRequest = this.tryGetPayload<ApiContracts.Website>();
+
+        const websiteDoc = await this.createOrUpdateWebsite(websiteRequest);
+
+        const pagesIterator = this.pageProvider.getPagesForWebsite(websiteDoc.id);
+        let websiteResponse = await this.convertWebsiteDocumentToResponse(websiteDoc, pagesIterator);
+
+        websiteResponse = await this.createNewPages(websiteResponse);
+
+        this.context.res = {
+            status: 200,
+            body: websiteResponse,
+        };
+        this.logger.logInfo('Website metadata successfully fetched from a storage.');
```

Comment:
do we save or read website here? is log message correct?
the status will be 201 if we creating here